platforms/jvm/toolchains-jvm-shared/src/main/java/org/gradle/jvm/toolchain/internal/JavaToolchainQueryService.java

```diff
@@ -236,12 +240,16 @@ private JavaToolchain downloadToolchain(JavaToolchainSpec spec, Set<JavaInstalla
     private JavaToolchain asToolchainOrThrow(InstallationLocation javaHome, JavaToolchainSpec spec, Set<JavaInstallationCapability> requiredCapabilities, boolean isFallback) {
         final JvmInstallationMetadata metadata = detector.getMetadata(javaHome);
 
-        if (!metadata.isValidInstallation()) {
+        if (!metadata.isValidInstallation() && !(spec instanceof SpecificExecutableToolchainSpec)) {
             throw new GradleException("Toolchain installation '" + javaHome.getLocation() + "' could not be probed: " + metadata.getErrorMessage(), metadata.getErrorCause());
         }
         if (!metadata.getCapabilities().containsAll(requiredCapabilities)) {
```

Comment:
❔ How does the capabilities check work in case of the specific executable? If the specific executable cannot be probed, doesn't that mean we'll always fail this check?

Reply:
Yeah, we'll fail the check if you request capabilities other than a launcher.  A `SpecificExectuableToolchainSpec` by definition returns a toolchain that can only be used for the `java` tool (i.e. a launcher).  If you provide one of these specs and ask for a different capability in the query (e.g. a compiler) this check will fail.  If you use `SpecificExectuableToolchainSpec` and don't specify any capabilities at query time (i.e. you only want a launcher), this check passes.  If you then try to ask the resultant toolchain for a different tool (e.g. `javac`) you'll get an error from the toolchain.